### PR TITLE
Updated Task Intro in ref doc

### LINKED ISSUE #2041: Revisit stream and task getting-started sections in the reference guide
As a user, I'm starting to deploy streams and task in SCDF, and I find the getting-started guides are inconsistent.

**Acceptance:**
- Similar to [Stream](https://docs.spring.io/spring-cloud-dataflow/docs/1.3.0.RELEASE/reference/htmlsingle/#spring-cloud-dataflow-stream-intro) section, we would have to put efforts to synchronize [Task](https://docs.spring.io/spring-cloud-dataflow/docs/1.3.0.RELEASE/reference/htmlsingle/#spring-cloud-dataflow-task) section in the reference guide
- Revisit the lifecycle section and review it for completeness

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #2041

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc [hunk 1]
```diff
@@ -9,18 +9,74 @@ It covers topics such as creating and running task applications.
...
+See http://cloud.spring.io/spring-cloud-dataflow/#platform-implementations[the Spring Cloud Data Flow project page] for a list of Data Flow server implementations.
+
+NOTE: Launching Spring Cloud Task applications can not be delegated to Skipper at this time.  You must use the Data Flow Server option to launch tasks.
+
+[[spring-cloud-dataflow-stream-intro-dsl]]
+=== Launching a task
+
+A task is defined by specifying the Task application you wish to execute along with any properties that you want to remain static for the definition.
+Each Spring Cloud Task application is registered under a simple name.
+The registration process specifies where the application can be obtained (for example, in a Maven Repository or a Docker registry).  You can find out more information on how to register Spring Cloud Task applications in this <<spring-cloud-dataflow-register-task-apps,section>>.
```
Comment (markpollack) [inline spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc]: Referring to the 'how to register apps' sections is too confusing.  In this getting started section, you should show how to register the timestamp task, just as we show how to register `http` source and `log` sink for streams.

**Comment by markpollack** [comment]
The `Deploying Tasks` section in the getting started area should link to the new section on getting started with tasks and not the section `Registering a Task Application`

File: spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc (hunk 1 above)
Comment (markpollack) [reply spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc]: Actually, this whole section for getting started should be moved up into the `getting-started.adoc` section.  

Look at https://docs.spring.io/spring-cloud-dataflow-server-cloudfoundry/docs/1.3.0.RELEASE/reference/htmlsingle/#getting-started-deploying-tasks

and 

https://docs.spring.io/spring-cloud-dataflow-server-kubernetes/docs/1.3.1.RELEASE/reference/htmlsingle/#_deploying_tasks

for reference

File: spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc [hunk 2]
```diff
@@ -9,18 +9,74 @@ It covers topics such as creating and running task applications.
...
 Tasks record the start time and the end time as well as the boot exit code in a relational database.
 The task implementation is based on the http://cloud.spring.io/spring-cloud-task/[Spring Cloud Task] project.
 
+The general lifecycle of a Task is:
+
+. Register Applications.
+. Create Task Definition.
+. Launch Task.
+
+You can use the Local Data Flow Server to launch tasks to your local machine, the Data Flow Server for Cloud Foundry to launch tasks to a single org and space on Cloud Foundry.
```
Comment (markpollack) [inline spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc]: As these docs are specific to getting started on local, I don't think we need to discuss the other platforms.

**Comment by cppwfs** [comment]
Updated based on code review comments.

**Comment by cppwfs** [comment]
I have updated the PR based on Code Review comments.   But it looks like it goes against what @sabbyanandan requested.

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc [hunk 3]
```diff
@@ -126,10 +126,10 @@ By default, the application registry is empty.
 As an example, register two applications, `http` and `log`, that communicate by using RabbitMQ.
 +
 ```
-dataflow:>app register --name http --type source --uri maven://org.springframework.cloud.stream.app:http-source-rabbit:1.2.0.RELEASE
+dataflow:>app register --name http --type source --uri maven://org.springframework.cloud.stream.app:http-source-rabbit:1.3.1.RELEASE
 Successfully registered application 'source:http'
 
-dataflow:>app register --name log --type sink --uri maven://org.springframework.cloud.stream.app:log-sink-rabbit:1.1.0.RELEASE
+dataflow:>app register --name log --type sink --uri maven://org.springframework.cloud.stream.app:log-sink-rabbit:1.3.1.RELEASE
```
Comment (markpollack) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc]: Updating this messes up the next getting started section for upgrading versions of the application with skipper....

**Comment by markpollack** [comment]
Added back docs on additional properties.  Added additional comments to look at task execution in getting started guide.  Other minor updates.

squashed and merged as 7e9351bcd1cfb96e733730707a332e3d647b39b8

Backported to 1.3.x as c1405308b5249c816987c7a956d23bc3905e0048

**Pull request closed without merging**